test-specifications/crypto-service/accountUpdateTransaction.md

```diff
@@ -19,111 +17,440 @@ https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_update.p
+|---------|---------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|-------------------|
+| 1       | Updates an account with a new valid ED25519 public key                                                  | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_ED25519_PUBLIC_KEY>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <CORRESPONDING_VALID_ED25519_PRIVATE_KEY>]                 | The account update succeeds and the account has the new ED25519 key.              | N                 |
+| 2       | Updates an account with a new valid ECDSAsecp256k1 public key                                           | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_ECDSA_SECP256K1_PUBLIC_KEY>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <CORRESPONDING_VALID_ECDSA_SECP256K1_PRIVATE_KEY>] | The account update succeeds and the account has the new ECDSAsecp256k1 key.       | N                 |
+| 3       | Updates an account with a new valid ED25519 private key                                                 | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_ED25519_PRIVATE_KEY>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <VALID_ED25519_PRIVATE_KEY>]                              | The account update succeeds and the account has the new ED25519 key.              | N                 |
+| 4       | Updates an account with a new valid ECDSAsecp256k1 private key                                          | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_ECDSA_SECP256K1_PRIVATE_KEY>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <VALID_ECDSA_SECP256K1_PRIVATE_KEY>]              | The account update succeeds and the account has the new ECDSAsecp256k1 key.       | N                 |
+| 5       | Updates an account with a new valid valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_KEYLIST>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <VALID_KEYLIST>]                                                      | The account update succeeds and the account has the new KeyList.                  | N                 |
+| 6       | Updates an account with a new valid KeyList of nested KeyLists (three levels)                           | accountId=<CREATED_ACCOUNT_ID>, key=<VALID_NESTED_KEYLIST>, commonTransactionParams.signers=[<PRIVATE_KEY_OF_CREATED_ACCOUNT>, <VALID_NESTED_KEYLIST>]                                        | The account update succeeds and the account has the new nested KeyList.           | N                 |
```

Comment:
Can we test minimum and maximum amount of keys in a KeyList?

Reply:
Not sure tbh, this needs checking

Reply:
In that case, I think that we can drop testing with the maximum number of keys in a KeyList. Tyvm for checking that out 👍🏻